### ISSUE TITLE
Unable to use two correlation properties for saga

### ISSUE THREAD

**Issue by poltaratskiy** [Particular/NServiceBus.Persistence.Sql#946]
Hello. According to this article https://docs.particular.net/samples/sql-persistence/transitioning-correlation-ids/ I try to use arguments correlationProperty and transitionalCorrelationProperty of SqlSaga attribute otherwise there's build errors appear:

```
[SqlSaga(
        tableSuffix: "TABLE",
        correlationProperty: nameof(SomeModel.RequestId),
        transitionalCorrelationProperty: nameof(SomeModel.CardId))]
```

In ConfigureHowToFindSaga:

```
protected override void ConfigureHowToFindSaga(SagaPropertyMapper<ReissueSagaModel> mapper)
{
        mapper.ConfigureMapping<FirstEvent>(message => message.RequestId)
            .ToSaga(sagaData => sagaData.RequestId);
        mapper.ConfigureMapping<SecondEvent>(message => message.CardId)
            .ToSaga(sagaData => sagaData.CardId);
}
```

And after start it throws exception with text "Sagas can only have mappings that correlate on a single saga property. Use custom finders to correlate ... ". 

In DB both columns CORR_REQUESTID and CORR_CARDID created.

**Comment by DavidBoike** [comment]
If you look closely at the [sample](https://docs.particular.net/samples/sql-persistence/transitioning-correlation-ids/) again in each phase, there's never 2 message mappings in ConfigureHowToFindSaga like you showed above.

Note that all the attribute does is makes sure that a column and a unique index is created for the transitional corrleation id.

So in each phase:

* [Phase 1](https://docs.particular.net/samples/sql-persistence/transitioning-correlation-ids/#phases-phase-1): Start with only having RequestId
* [Phase 2](https://docs.particular.net/samples/sql-persistence/transitioning-correlation-ids/#phases-phase-2): Add attribute so that CardId column is created and start filling it with data as each message comes through. You have to wait until ALL rows in the database have a CardId populated before continuing. The ConfigureHowToFindSaga method is still mapping based on RequestId.
* [Phase 3](https://docs.particular.net/samples/sql-persistence/transitioning-correlation-ids/#phases-phase-3): The mapping changes to CardId. The attribute can be removed, and the CORR_REQUESTID column can be removed from the table.

In no situation do you get to map 2 different message types with different values. If both `FirstEvent` and `SecondEvent` are involved in the saga, they both need to have the `RequestId` and `CardId` properties, at least for the duration of the transition.

**Comment by poltaratskiy** [comment]
Thank you for answer. I didn't see that unique index is created for transitional correlation id.

I saw the phase 2 and decided to try it. 

And I have misunderstanding why it's impossible to find saga by multiple correlation ids, I've faced that situation twice, first time I created separated saga, this time external system sends data with another id (CardId in my case) and I need to find saga by another id. 
I saw [custom finders](https://docs.particular.net/samples/saga/sql-sagafinder/) but I wanted to get easier way to solve my issue.
